src/internal/observe/observe.go

```diff
@@ -239,15 +239,15 @@ func ItemProgress(
 	header string,
 	iname any,
 	totalBytes int64,
-) (io.ReadCloser, func()) {
+) (io.ReadCloser, func(), func()) {
```

Comment:
mmm so I know it's a bit more work, but this may be best addressed in a separate PR that goes in prior to this one. I think my biggest concern about returning both an abort and a done function centers around what happens if abort is called and then done is called (i.e. we deferred the done call but ended up with an error and called abort instead)?

There's also some WaitGroup usage in this file that may not be addressed by the abort call which could potentially cause deadlocks if we're not careful

Reply:
Second this.  We can stick with the current implementation with itemProgress display duplication (maybe hold off on this pr merger until after our next release, if it's on monday?) and fix up the situation after.

Also of note: having a closer func for observe is already a bit heavy.  Ideally, this interface would work similar to a log, where you can fire and forget.  Unfortunately, we don't have exactly that, so it goes, we can make do with a closer.  Beyond that, interaction starts to get really unwieldy.  A clean and minimized interface is important here.

That's not to say we can't have an abort func (validating ash's worries notwithstanding).  But we'll probably want it in the form of returning a struct representing the display persistence, and have multiple control methods available through that.  Or something of that nature.

Reply:
I don't think this will be an issue. What we return in `closer` is [this](https://github.com/alcionai/corso/blob/d00696621f545ea25098f5f69348146bc837f419/src/internal/observe/observe.go#L432), which is essentially wait for the bar to end(abort or complete) , then remove the wg item and print log. That function does not really mark it as "done". It just waits for a done message to do some cleanup and logging. When we call abort, we just happen to simulate the "mark as done" action and stop the bar.

Reply:
I've created https://github.com/alcionai/corso/pull/3535 where we completely avoid returning closers to the consumer and call that within the progressbar creation code. This should eliminate any potential issues with progressbars.

That PR can be merged into this before this goes into main.

Reply:
Now that https://github.com/alcionai/corso/pull/3535 is merged, I would like another round of review @ashmrtn @ryanfkeepers

Reply:
Thanks for the PR on progress bars. I don't mean to unduly hold up this PR or belabor the point, but unfortunately locking/deadlocks is one of the things that are best viewed with as few other changes possible since getting it wrong can be painful and sometimes difficult to reproduce

Reply:
No worries. I would rather fix this now when we know the issue than later hunting in the dark when something breaks :D